src/09-led-compass/magnitude.md

```diff
@@ -1,67 +1,85 @@
 # Magnitude
 
-We have been working with the direction of the magnetic field but what's its real magnitude? The
-number that the `magnetic_field` function reports are unit-less. How can we convert those values to
-Gauss?
-
-The documentation will answer that question.
-
-> Section 2.1 Sensor characteristics - Page 10 - LSM303DLHC Data Sheet
-
-The table in that page shows a *magnetic gain setting* that has different values according to the
-values of the GN bits. By default, those GN bits are set to `001`. That means that magnetic gain of
-the X and Y axes is `1100 LSB / Gauss` and the magnetic gain of the Z axis is `980 LSB / Gauss`. LSB
-stands for Least Significant Bits and the `1100 LSB / Gauss` number indicates that a reading of
-`1100` is equivalent to `1 Gauss`, a reading of `2200` is equivalent to 2 Gauss and so on.
-
-So, what we need to do is divide the X, Y and Z values that the sensor outputs by its corresponding
-*gain*. Then, we'll have the X, Y and Z components of the magnetic field in Gauss.
-
-With some extra math we can retrieve the magnitude of the magnetic field from its X, Y and Z
-components:
+We have been working with the direction of the magnetic field but what's its real magnitude?
+According to the documentation about the [`mag_data()`] function the x y z values we are
+getting are already in milli-gauss. That means the only thing we have to compute in order
+to get the magnitude of the magnetic field is the magnitude of the 3D vector that our x y z
+values describe. As you might remember from school this is simply:
 
 ``` rust
-let magnitude = (x * x + y * y + z * z).sqrt();
+let magnitude = sqrtf(x * x + y * y + z * z);
```

Comment:
Maybe we should explain somewhere that we get use the `sqrt` function from `libm`?